C error when using `try_push` fn on channel.

### Describe the bug

C error is generated when using the `try_push` function on a channel.

### Reproduction Steps

```v
fn main() {
    foo := chan string{}
    foo.try_push('some string')
}
```

### Expected Behavior

Code should have compiled.

### Current Behavior

C error is generated:

```
================== C compilation error (from cc): ==============
cc: /tmp/v_501/test.01JFRMRA0873XNTVKDVDDZ9XEP.tmp.c:9277:39: error: operand of type 'string' (aka 'struct string') where arithmetic or pointer type is required
cc:  9277 |         sync__Channel_try_push(foo, (voidptr)_SLIT("some string"));
cc:       |                                              ^~~~~~~~~~~~~~~~~~~~
cc: /tmp/v_501/test.01JFRMRA0873XNTVKDVDDZ9XEP.tmp.c:218:18: note: expanded from macro '_SLIT'
cc:   218 | #define _SLIT(s) ((string){.str=(byteptr)("" s), .len=(sizeof(s)-1), .is_lit=1})
cc:       |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc: 1 error generated.
================================================================
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.9 2911f29.98f13ef

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.9 2911f29.98f13ef
OS: macos, macOS, 15.1.1, 24B91
Processor: 8 cpus, 64bit, little endian, Apple M3

getwd: /Users/adamoates/Documents/shattlebip
vexe: /Users/adamoates/v/v
vexe mtime: 2024-12-23 02:11:42

vroot: OK, value: /Users/adamoates/v
VMODULES: OK, value: /Users/adamoates/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.39.5 (Apple Git-154)
Git vroot status: weekly.2024.50-100-g98f13ef4
.git/config present: true

CC version: Apple clang version 16.0.0 (clang-1600.0.26.4)
emcc version: N/A
thirdparty/tcc status: thirdparty-macos-arm64 713692d4
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzY4YzgxOGYxNDRmNTg1YWU0ZWNkZGQiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.K-Rqj6z8HOzkW9fpCrxg_2qJjbsIORENkUffPXWG8Xg">Huly&reg;: <b>V_0.6-21676</b></a></sub>

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2924,7 +2924,10 @@ fn (mut g Gen) ref_or_deref_arg(arg ast.CallArg, expected_type ast.Type, lang as
 						g.expr_with_opt(arg.expr, arg_typ, expected_type)
 						return
 					} else if arg.expr.is_literal() {
-						g.write('(voidptr)')
+						g.write('(voidptr)ADDR(${g.styp(arg_typ)}, ')
+						g.expr(arg.expr)
+						g.write(')')
+						return
 					} else {
 						if arg_typ_sym.kind in [.sum_type, .interface] {
 							atype = arg_typ

--- a/vlib/v/tests/concurrency/chan_try_push_literal_test.v
+++ b/vlib/v/tests/concurrency/chan_try_push_literal_test.v
@@ -1,0 +1,5 @@
+fn test_chan_try_push_literal() {
+	foo := chan string{}
+	foo.try_push('some string')
+	assert true
+}

